diff -Nau2r trac.orig/trac/ticket/notification.py trac.new/trac/ticket/notification.py
--- trac.orig/trac/ticket/notification.py	2015-10-09 20:23:39.926804169 +0200
+++ trac.new/trac/ticket/notification.py	2015-10-10 17:27:20.512955684 +0200
@@ -479,5 +479,9 @@
         }
         if not self.newticket:
-            msgid = self.get_message_id(dest)
+            # Hack for Trac 1.0 to make trac-messageid-plugin work:
+            rows = self.env.db_query("""
+                    SELECT messageid FROM messageid WHERE ticket=%s
+                    """, (self.ticket.id,))
+            msgid = rows[0][0] if rows else self.get_message_id(dest)
             hdrs['In-Reply-To'] = msgid
             hdrs['References'] = msgid
